[Johnwz123] iP

Johnwz123 opened the pull request:
# Pull Request Heading

## Bullet list
- Bullet 1
- Bullet 2
- Bullet 3

## Numbered list
1. Number 1
1. Number 2
1. Number 3

## Fenced code block (with syntax highlighting)
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

## Task list
- [ ] Task 1
- [ ] Task 2
- [ ] Task 3

## Others
An emoji 😄 
> A blockquote

[A hyperlink](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
Some `inline` code
some text formatting: **bold**, _italic_, ~strikethrough~ etc.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Johnwz123 committed:
Rename chatbot

Johnwz123 committed:
Implement initial skeleton

Johnwz123 committed:
Implement echo command and exit functionality

Johnwz123 committed:
Implement add and list functionality

Johnwz123 committed:
Create Task class

Johnwz123 committed:
Add feature to mark tasks as done or not done

Johnwz123 committed:
Create ToDo, Deadline and Event class

Johnwz123 committed:
Add support for tracking ToDos, Events and Deadlines

Johnwz123 committed:
Update tests

Johnwz123 committed:
Add error handling for commands

Johnwz123 committed:
Add feature for deleting tasks

Johnwz123 committed:
Add GitHub Copilot acknowledgement

Johnwz123 committed:
Refactor to use enums for task type

Johnwz123 committed:
Add ability to save and load from hard disk

Johnwz123 committed:
Merge branch 'branch-Level-7'

Johnwz123 committed:
Add date and time parsing

Johnwz123 committed:
Merge branch 'branch-Level-8'

Johnwz123 committed:
Refactor code

Johnwz123 committed:
Organize classes into packages

Johnwz123 committed:
Remove unused import statement

Johnwz123 committed:
Merge branch 'add-gradle-support'

Johnwz123 committed:
Add tests

Johnwz123 committed:
Add JavaDoc comments

Johnwz123 committed:
Reformat code

Johnwz123 committed:
Add search tasks functionality

Johnwz123 committed:
Create package for Buddy class

Johnwz123 committed:
Add javadoc comments

Johnwz123 committed:
Merge branch 'branch-A-JavaDoc'

Johnwz123 committed:
Merge branch 'branch-A-CodingStandard'

Johnwz123 committed:
Add javadoc comments

Johnwz123 committed:
Fix style

Johnwz123 committed:
Setup Java FX

Johnwz123 committed:
Link Buddy to JavaFx

audipras commented on src/main/java/command/InvalidCommand.java:
I like that an instance of InvalidCommand is returned in errors! This makes error handling intuitive and maintainable in the future. Well done!

audipras commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private static final DateTimeFormatter OUTPUT_FORMATTER = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
```
INPUT_FORMATTER and OUTPUT_FORMATTER are apt names for this constants. Well done!

audipras commented on src/main/java/buddy/MainWindow.java:
```diff
@@ -0,0 +1,66 @@
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField userInput;
+    @FXML
+    private Button sendButton;
```
I like the naming of your Ui components - very intuitive!

audipras commented on src/main/java/buddy/Launcher.java:
```diff
@@ -0,0 +1,12 @@
+import javafx.application.Application;
+
+/**
+ * A launcher class to workaround classpath issues.
```
Perhaps putting this in a comment as opposed to javadocs is more appropriate?

audipras reviewed (commented):
LGTM! The code follows the coding standards precisely, making it readable and maintainable. Very few nits (if any), overall excellent job!

SamuelFoo commented on src/main/java/buddy/Buddy.java:
```diff
@@ -0,0 +1,73 @@
+        this.storage = new Storage(filePath);
+        TaskList loadedTasks;
+        try {
+            loadedTasks = new TaskList(storage.load());
```
Good use of OOP to handle both the reading of stored data and task list!

SamuelFoo commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,116 @@
+    /**
+     * Returns a message when a task is unmarked.
+     */
+    public String showTaskUnmarked(Task task) {
```
Maybe handling the response messages could be added to the individual command classes instead? The command classes could provide the response text and then call a function in the UI to display the message instead.

SamuelFoo commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,79 @@
+            return new ListCommand();
+        case "todo":
+            return new AddCommand(new ToDo(arguments));
+        case "deadline":
```
Again a matter of opinion. Maybe the exception handling could be done in the individual command classes?  `Parser` could still handle the splitting and arguments though.

SamuelFoo commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+ * Represents an event task.
+ */
+public class Event extends Task {
+    private static final DateTimeFormatter INPUT_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```
Great job with the date-time formatting!

SamuelFoo commented on src/test/java/parser/ParserTest.java:
```diff
@@ -0,0 +1,101 @@
+
+public class ParserTest {
+    @Test
+    public void testParseValidListCommand() {
```
A very minor point. But maybe to be safe you could follow the naming convention for testing functions here: https://se-education.org/se-book/cppToJava/junit/basic/index.html

SamuelFoo reviewed (commented):
LGTM! Great job completing the tasks! Sorry if I am overly pedantic. Thanks.

Johnwz123 committed:
Add assertions to TaskList and Buddy classes

The TaskList and Buddy classes lack assertions to ensure the internal state is consistent and correct during execution.

Assertions are needed to document important assumptions and catch potential issues early in the development process.

Assertions are added this way to catch potential issues early and document important assumptions, improving code reliability and maintainability.

Johnwz123 committed:
Improve naming

Johnwz123 committed:
Merge pull request #1 from Johnwz123/branch-A-Assertions

Add assertions to TaskList and Buddy classes

Johnwz123 committed:
Merge branch 'master' into branch-A-CodeQuality

Johnwz123 committed:
Merge pull request #2 from Johnwz123/branch-A-CodeQuality

Improve naming

Johnwz123 committed:
Set up Continuous Integration (CI)

The project currently lacks a Continuous Integration (CI) setup, which is essential for automating the build and testing process.

Setting up CI is necessary to ensure that the codebase remains stable and that new changes do not introduce regressions.

Set up Continuous Integration (CI) for the project

The project currently lacks a Continuous Integration (CI) setup, which is essential for automating the build and testing process.

Setting up CI is necessary to ensure that the codebase remains stable and that new changes do not introduce regressions.

Johnwz123 committed:
Fix formatting

Johnwz123 committed:
Fix StorageTest

Johnwz123 committed:
Update runtest.sh

Johnwz123 committed:
Update CI workflow

Johnwz123 committed:
Add archive feature

Johnwz123 committed:
Add Javadoc comment for ArchiveCommand class

Johnwz123 committed:
Add title to application

Johnwz123 committed:
Add close application functionality

Johnwz123 committed:
Add user guide
